### PR TITLE
[WIP] [DO NOT REVIEW] Pf additional latency

### PR DESCRIPTION
**Pull request by wojtek-t**

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go [hunk 1]
```diff
@@ -475,6 +485,10 @@ func (qs *queueSet) chooseQueueIndexLocked(hashValue uint64, descr1, descr2 inte
 	bestQueueLen := int(math.MaxInt32)
 	// the dealer uses the current desired number of queues, which is no larger than the number in `qs.queues`.
 	qs.dealer.Deal(hashValue, func(queueIdx int) {
+
+		// FIXME: This isn't the right way to choose the queue anymore.
+		// Instead of taking length() we have to take sum of widths.
+		// And probably somehow accomodate `additional latency` too.
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go:491]: @tkashem - this is one more thing has to be updated.

File: staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go [hunk 2]
```diff
@@ -337,6 +337,16 @@ func (req *request) Finish(execFn func()) bool {
 		}()
 
 		execFn()
+
+		// Sleep for additional latency to block concurrency units.
+		if req.additionalLatency > 0 {
+			// TODO: In the current form this means that it's actually contributing to the
```
Comment (tkashem) [inline staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go:343]: @wojtek-t  yes, the additional latency should not have any impact on the user experience, i have a work in progress where we can achieve this without having any impact on user experience - https://github.com/kubernetes/kubernetes/pull/103240

The second commit https://github.com/kubernetes/kubernetes/pull/103240/commits/2a23c504622ffed2011c4cbf4997ca249338c665 sleeps outside the goroutine that executes the request handler.

The first commit make room for additional latency into width - i can remove it if we don't need it

Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go:341]: Great - I commented on that PR, it looks cool.

**Comment by wojtek-t** [comment]
Closing in favor of bunch of other PRs (parts of it are merged/merging, additional -latency has a separate PR by Abu).

**Pull request closed without merging**